packages/kit/src/core/adapt/utils.js

```diff
@@ -47,6 +48,60 @@ export function get_utils({ cwd, config, build_data, log }) {
 					log
 				});
 			}
+		},
+
+		/** @param {{patterns: string[], generate?: boolean}} options */
+		update_ignores({ patterns, generate = false }) {
```

Comment:
this seems quite complicated, perhaps there is a reason, but I can't figure out why we need to be so careful to preserve formatting of the existing file

Wouldn't a better solution to read the whole file, split by line break, insert into a Set (to make them unique) and then `[ ...mySet ].join(NEWLINE_CHAR)`

Reply:
I'm being careful because I usually wouldn't want a tool to just reorganize all the lines inside my .gitignore by alphabetical order

Reply:
I retire what I said above, javascript sets preserve the insertion order. But the position of the new lines should be inside a "block" to make it clear to the user where the changes were made

Reply:
You could add a `# comment line` before and after the inserted lines to indicate that they were generated.

But lets be clear that we're talking about an operation that would normally be done just before deployment, on a set of files which are generated just for that deployment. We're not editing the source. The whole file is in a "generated" dir, isn't it?